src/tree-internal.component.ts

```diff
@@ -61,13 +64,19 @@ export class TreeInternalComponent implements OnInit {
   public isSelected: boolean = false;
   public isRightMenuVisible: boolean = false;
   public isLeftMenuVisible: boolean = false;
+  public api: TreeInternalAPI;
 
   public constructor(@Inject(NodeMenuService) private nodeMenuService: NodeMenuService,
                      @Inject(TreeService) public treeService: TreeService,
                      @Inject(ElementRef) public element: ElementRef) {
   }
 
   public ngOnInit(): void {
+    this.api = new TreeInternalAPI(this);
```

Comment:
Let's name it TreeController instead of TreeInternalAPI, cause it controls a tree - allows you to select, expand, collapse ...